app/org/sagebionetworks/bridge/services/StudyService.java

```diff
@@ -259,23 +284,32 @@ public Study updateStudy(Study study, boolean isAdminUpdate) {
         
         return updatedStudy;
     }
-    
-    public void deleteStudy(String identifier) {
+
+    public void deleteStudy(String identifier, boolean physical) {
         checkArgument(isNotBlank(identifier), Validate.CANNOT_BE_BLANK, "identifier");
 
         if (studyWhitelist.contains(identifier)) {
             throw new UnauthorizedException(identifier + " is protected by whitelist.");
         }
 
-        // Verify the study exists before you do this.
-        Study existing = getStudy(identifier);
+        // Verify the study exists before you do this
+        // only admin can call this method, should contain deactivated ones.
+        Study existing = getStudy(identifier, true);
         if (existing == null) {
             throw new EntityNotFoundException(Study.class, "Study '"+identifier+"' not found");
         }
-        studyDao.deleteStudy(existing);
-        directoryDao.deleteDirectoryForStudy(existing);
-        subpopService.deleteAllSubpopulations(existing.getStudyIdentifier());
-        cacheProvider.removeStudy(identifier);
+
+        if (!physical) {
+            // deactivate
+            if (!existing.isActive()) throw new EntityNotFoundException(Study.class, "Study '"+identifier+"' is deactivated before.");
+            studyDao.deactivateStudy(existing.getIdentifier());
```

Comment:
I think you need to remove it from the cache as well. It might show up in certain calls if you don't, not sure